lib/promscrape/discovery/kubernetes/endpoints.go

```diff
@@ -194,6 +204,19 @@ func getEndpointLabelsForAddressAndPort(gw *groupWatcher, podPortsSeen map[*Pod]
 			}
 		}
 	}
+	for _, c := range p.Spec.InitContainers {
+		// Defines native sidecar https://kubernetes.io/blog/2023/08/25/native-sidecar-containers/#what-are-sidecar-containers-in-1-28
+		if c.RestartPolicy != "Always" {
```

Comment:
Btw, prometheus adds unconditionally `initContainers` to the `Containers` and I think it's a bad behavior. Your solution is better to me.